travellocal/amqp-wrapper
# Bring TL version of mq-rabbit up to speed and resolve channel closing

This PR covers quite a lot of ground, with the goals of:

- Bringing the repo into 2018 in terms of dependencies, type definitions, Node version etc
- Tidying up the tests
- Fixing #1 - this was the main reason to create the fork

Closes #1 
Closes #2, #4, #5 

## QA

Mainly this should just be a code review, but it would also be a good idea to clone the repo and run the tests. They require a running RabbitMQ instance - if your https://github.com/travellocal/travellocal RabbitMQ is running on `localhost` they should work fine. In future it would be good to either mock out the RabbitMQ instance or add a Docker container here that spins up the instance for test purposes.

Things to check:

- [x] the library builds correctly
- [x] messages are published correctly
- [x] messages are consumed correctly
- [x] channels are closed correctly once messages have been published

Separately, we should decide how we're going to use this dependency. Should we install directly from GitHub? Or should we use a private NPM repository? We could also send these changes upstream to the RM version but I don't want to wait on that / it might be better to keep this specific to the TravelLocal use-case.
